lib/gollum-lib/file.rb

```diff
@@ -2,55 +2,103 @@
 
 module Gollum
   class File
+
+    class << self
+      # For use with self.find: returns true if the given query corresponds to the in-repo path of the BlobEntry. 
+      #
+      # query     - The String path to match.
+      # entry     - The BlobEntry to check against.
+      def path_match(query, entry)
+        query == ::File.join('/', entry.path)
+      end
+    end
+
+    # Find a file in the given Gollum wiki.
+    #
+    # wiki    - The wiki.
+    # path    - The full String path.
+    # version - The String version ID to find.
+    # try_on_disk - If true, try to return just a reference to a file
+    #               that exists on the disk.
+    #
+    # Returns a Gollum::File or nil if the file could not be found. Note
+    # that if you specify try_on_disk=true, you may or may not get a file
+    # for which on_disk? is actually true.
+    def self.find(wiki, path, version, try_on_disk = false)
+      map = wiki.tree_map_for(version.to_s)
+
+      if wiki.page_file_dir
```

Comment:
I think here the ternary operator would do nicely:
```ruby
query_path = wiki.page_file_dir ? ::File.join(wiki.page_file_dir, path) : path
```
Sweet and short, and still understandable!

Reply:
Excellent!